src/FunctionControlPanel.java

```diff
@@ -0,0 +1,199 @@
+    private final JPanel functionInputPosition;
+    private final PolynomialFunctionControl polynomialFunctionControl;
+
+    JTextField functionField = new JTextField(20);  //Textfield for the Function
+
+    public FunctionControlPanel(int count, List<FunctionControlPanel> functionControlPanelList, JPanel functionInputPosition, PolynomialFunctionControl polynomialFunctionControl) {
+        this.functionControlPanelList = functionControlPanelList;
+        this.functionInputPosition = functionInputPosition;
+        this.count = count;
+        this.polynomialFunctionControl = polynomialFunctionControl;
+        this.setBackground(Color.blue);
+
+
+        gbc.fill = GridBagConstraints.BOTH; //Sets the Object to fill the Horizontal Space
+        gbc.insets = new Insets(4, 1, 4, 1); //Sets the distance between the Objects
+        setLayout(InputLayout);
+
+        createFunctionField();
+        createColorPicker();
+        createCalculateButton();
+        createDeriveButton();
+        createDeleteButton();
+    }
+
+    private class ColorBoxListener implements ActionListener {
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            Color selectedColor = (Color) colorPicker.getSelectedItem(); //Gets the selected Color from the ComboBox (cast to Color)
+            System.out.println("Selected color: " + selectedColor); //Test if the selected Color is correct
+            //TODO: Connect to the function color "setter"
+        }
+    }
+
+    private class CalculateActionListener implements ActionListener {
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            //System.out.println(functionField.getText());
+            //TODO: Connect to the function calculator
+            System.out.println("Calculate from: " + count);
+        }
+    }
+
+    public class DeriveActionListener implements ActionListener {
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            //TODO: Connect to the function deriver
+        }
+    }
+
+    private class DeleteActionListener implements ActionListener {
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            removeFunctionControlPanel(FunctionControlPanel.this);
+            //TODO: Connect to the function deleter to delete function on the GraphPanel
+        }
+    }
+    private void removeFunctionControlPanel(FunctionControlPanel functionControlPanel) {
+
+        if (functionControlPanelList.size() > 1) {
+            functionInputPosition.remove(functionControlPanel);
+            functionControlPanelList.remove(functionControlPanel);
+            polynomialFunctionControl.revalidate();
+            polynomialFunctionControl.repaint();
+            PolynomialFunctionControl.decrementCount();
+            count--;
+            System.out.println("Delete");
+        }
+    }
+
+    private class ColorListRenderer extends DefaultListCellRenderer {
```

Comment:
Maybe we can make use of JColorChooser:
https://docs.oracle.com/javase/tutorial/uiswing/components/colorchooser.html

(Optional, nice to have, but not required.)

Reply:
As far as i know he tried it but it made things complicated. Would put this maybe on hold for a future "feature"?

Reply:
Done